docs/api-reference/geo-layers/tile-layer.md

```diff
@@ -187,16 +187,35 @@ The maximum memory used for caching tiles. If this limit is supplied, `getTileDa
 - Default: `null`
 
 
-##### `refinementStrategy` (Enum, optional)
+##### `refinementStrategy` (String|Function, optional)
 
-How the tile layer refines the visibility of tiles. One of the following:
+How the tile layer refines the visibility of tiles. When zooming in and out, if the layer only shows tiles from the current zoom level, then the user may observe undesirable flashing while new data is loading. By setting `refinementStrategy` the layer can attempt to maintain visual continuity by displaying cached data from a different zoom level before data is available.
+
+This prop accepts one of the following:
 
 * `'best-available'`: If a tile in the current viewport is waiting for its data to load, use cached content from the closest zoom level to fill the empty space. This approach minimizes the visual flashing due to missing content.
 * `'no-overlap'`: Avoid showing overlapping tiles when backfilling with cached content. This is usually favorable when tiles do not have opaque backgrounds.
 * `'never'`: Do not display any tile that is not selected.
+* A custom function. See "custom strategy" below.
 
 - Default: `'best-available'`
 
+##### custom strategy
+
+Apps may defin a custom `refinementStrategy` by supplying its own callback function. The function will be called frequently on every viewport update and every tile loaded event.
+
+When called, the function receives an array of `Tile` instances representing every tile that is currently in the cache. Each tile is populated with the following properties:
+  - `x` (Number) - x index of the tile
+  - `y` (Number) - y index of the tile
+  - `z` (Number) - z index of the tile
+  - `parent` (Tile) - the parent tile (a tile on `z-1` that contains this tile), if present in the cache
+  - `children` (Tile[]) - the sub tiles (tiles on `z+1` that are contained by this tile), if present in the cache
+  - `isSelected` (Boolean) - if the tile is expected to show up in the current viewport
+  - `isVisible` (Boolean) - if the tile should be rendered
+  - `isLoaded` (Boolean) - if the content of the tile has been loaded
+
+The callback is an opportunity to manipulate `isVisible` before sub layers are rendered. `isVisible` is initially set to the value to `isSelected` (equivalent to `refinementStrategy: 'never'`).
```

Comment:
Something we want to experiment with is creating temporary dummy placeholder children from parent data, so it would be nice to be able to do that in this callback.

Reply:
I have not tried but it should be possible here by manually writing into `tile.content`. `tile.content` is what is passed to sub layers' `data` prop. When loading is completed `tile.content` would be overwritten by actual data.

Reply:
However, arguably this is not the right place to create dummy data (you only need to do it once while this callback is called often). Alternatively we could offer an `onTileRequest` callback which is called when a new tile is requested. You will be able to change the default value of `tile.content` then.

Reply:
OK, we can try that.